### PR TITLE
fix(groupby): stabilize empty agg/apply results (#127)

### LINKED ISSUE #127: A2: Stabilize GroupBy aggregations/apply for empty-group results
Parent roadmap: #125

## Problem
`GroupBy.agg()` / `GroupBy.apply()` currently call `DataFrame.concat(results)`. For empty group output, `results=[]` and `concat` throws, causing empty-input grouped operations to fail.

## Scope
1. Make grouped operations deterministic for zero-row/zero-group inputs.
2. Return correctly shaped empty DataFrames instead of throwing.

## Proposed implementation
1. Add explicit empty-result guard before concat in `GroupBy.agg()` and `GroupBy.apply()`.
2. Build expected output schema for empty agg results (keys + planned output columns).
3. Preserve index semantics with existing API conventions.

## Acceptance criteria
- `groupBy(...).agg(...)` on empty DataFrame returns empty DataFrame with expected columns.
- `groupBy(...).apply(...)` on empty DataFrame returns empty DataFrame (no throw).
- Non-empty grouped behavior remains unchanged.

## Tests
- Local regressions for empty/non-empty grouped agg/apply.
- Column/schema assertions for empty result outputs.

### PR DESCRIPTION
**Pull request by joe-broadhead**
## Summary
- fix `GroupBy.agg(...)` and `GroupBy.apply(...)` for empty-input scenarios
- return deterministic empty DataFrames instead of throwing via `DataFrame.concat([])`
- add local regression tests for empty agg/apply behavior and schema stability

Closes #127.

## Details
- `agg(...)` now returns an empty DataFrame with expected output schema (`group keys + agg output columns`) when no groups are present.
- `apply(...)` now returns an empty DataFrame that preserves source columns when no groups are present.
- Existing non-empty behavior remains unchanged.

## Validation
- `npm run lint`
- `npm run test:local`
- `npm run test:perf:check`

### CODE REVIEW

**Comment by joe-broadhead** [comment]
@codex review

File: apps_script_tools/groupBy/GroupBy.js [hunk 1]
```diff
@@ -101,11 +116,29 @@ var GroupBy = class GroupBy {
...
-  
+
+      if (results.length === 0) {
+        const emptyColumns = {};
+        const emptyOutputColumns = this.keys.concat(outputPlans.map(plan => plan.outputName));
+
+        for (let idx = 0; idx < emptyOutputColumns.length; idx++) {
+          const columnName = emptyOutputColumns[idx];
+          if (!Object.prototype.hasOwnProperty.call(emptyColumns, columnName)) {
+            emptyColumns[columnName] = Series.fromArray([], columnName);
```
[the remark this answers is not in the document]
Comment (joe-broadhead) [reply apps_script_tools/groupBy/GroupBy.js]: Agreed and fixed. The empty `agg(...)` path now preserves schema metadata instead of defaulting to `undefined`:

- Group key columns are rebuilt from source series with original `name`, `type`, and `useUTC`.
- Aggregation output columns now get deterministic type assignment via reducer-aware inference (`number`/`boolean` where applicable, otherwise source type fallback).
- Added regression schema test in `tests/local/groupby-empty-results.test.mjs` (`GroupBy.agg preserves key and output schema types for empty typed input`).

Included in commit `307d675`.
